### ISSUE TITLE
creating new formula 'scirius' (rules manager for 'suricata' IDPS)

### ISSUE THREAD

**Issue by Drewshg312** [Homebrew/homebrew#41499]
Hi! I've found and tested awesome rule manager for Suricata IDPS. It would be a great addition to Elasticsearch+Logstash+Kibana+Suricata in homebrew. Scirius only available from debian packages so far. So I've tried to create my first homebrew formula... But I have no experience in ruby... So I want to contribute but I need help and direction (to make it properly)... It shouldn't be that hard...

How I installed it on my mac:
https://github.com/StamusNetworks/scirius

INSTALLATION:
Clone the source to the place you want:

          $ git clone https://github.com/StamusNetworks/scirius /usr/local/var/www/scirius

Upgrade pip before installing deopendencies:

          $ pip install --upgrade pip

Install dependencies:

          $ cd /usr/local/var/www/scirius
          $ pip install -r requirements.txt

To use the suri_reloader script which is handling suricata restart, you will also need pyinotify
(Not available for Yosemite and suri_reloader is going to be rewritten for mac os probably):

          $ pip install pyinotify

You will also potentially needs the gitdb module
(if not already installed):

          $ pip install gitdb

START

 Get the source, then inside the source:

        $ python /usr/local/var/www/scirius/manage.py syncdb

Authentication is by default in scirius so you will need to create a superuser account when proposed.
One of the easiest way to try Scirius is to run the Django test server

        $ python /usr/local/var/www/scirius/manage.py runserver

You can then connect to localhost:8000.

If you need the application to listen to a reachable address, you can run something like

        $ python manage.py runserver 192.168.1.1:8000

**Comment by Drewshg312** [comment]
what I've done so far:
    
      $ brew create https://github.com/StamusNetworks/scirius

and then I've modified the template (adding version, dependancy 'python' and plist)

**scirius.rb:**

    # Documentation: https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/Formula-Cookbook.md
    #                /usr/local/Library/Contributions/example-formula.rb
    # PLEASE REMOVE ALL GENERATED COMMENTS BEFORE SUBMITTING YOUR PULL REQUEST!
    
    class Scirius < Formula
      desc "Rules manager for Suricata IDPS"
      homepage "https://github.com/StamusNetworks/scirius"
      url "https://github.com/StamusNetworks/scirius"
      version "1.0."
      sha256 ""
    
      # depends_on "cmake" => :build
      #depends_on :x11 # if your formula requires any X11/XQuartz components
      depends_on :python
    
      def install
        # ENV.deparallelize  # if your formula fails when building in parallel
    
        # Remove unrecognized options if warned by configure
        system "./configure", "--disable-debug",
                              "--disable-dependency-tracking",
                              "--disable-silent-rules",
                              "--prefix=#{prefix}"
        # system "cmake", ".", *std_cmake_args
        system "make", "install" # if this fails, try separate make/make install steps
      end
    
    plist_options :manual => "python"
    
      def plist; <<-EOS.undent
        <?xml version="1.0" encoding="UTF-8"?>
        <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
        <plist version="1.0">
            <dict>
                <key>KeepAlive</key>
                <true/>
                <key>Label</key>
                <string>#{plist_name}</string>
                <key>ProgramArguments</key>
                <array>
                    <string>#{HOMEBREW_PREFIX}/bin/python</string>
                    <string>/usr/local/var/www/scirius/manage.py</string>
                    <string>runserver</string>
                </array>
                <key>RunAtLoad</key>
                <true/>
            </dict>
        </plist>
      EOS
      end
    
    
    
      test do
        # `test do` will create, run in and delete a temporary directory.
        #
        # This test will fail and we won't accept that! It's enough to just replace
        # "false" with the main program this formula installs, but it'd be nice if you
        # were more thorough. Run the test with `brew test scirius`. Options passed
        # to `brew install` such as `--HEAD` also need to be provided to `brew test`.
        #
        # The installed folder is not in the path, so use the entire path to any
        # executables being tested: `system "#{bin}/program", "do", "something"`.
        system "false"
      end
    end

**Comment by mikemcquaid** [comment]
@Drewshg312 Could you try and open a pull request? This should help and I'm happy to walk you through anything else: https://github.com/Homebrew/homebrew/blob/master/share/doc/homebrew/How-To-Open-a-Homebrew-Pull-Request-(and-get-it-merged).md#how-to-open-a-homebrew-pull-request-and-get-it-merged

Thanks!

**Comment by Drewshg312** [comment]
Done! Could you confirm if everything right?

**Comment by Drewshg312** [comment]
well I've moved a little bit further and now I got this:

    class Scirius < Formula
      desc "Rules manager for Suricata IDPS"
      homepage "https://github.com/StamusNetworks/scirius"
      url "https://github.com/StamusNetworks/scirius/archive/scirius-1.0.tar.gz"
      sha256 "fde16b3ac8f95e2990a44a9d85aa7c3ee6b55e59af9c27ded7319db19c19b2f5"
    
      depends_on "python"
    
      resource "daemon" do
        url "https://pypi.python.org/packages/source/d/daemon/daemon-1.1.tar.gz"
        sha1 "520de0f52875417da165ff05167c33e6cdfb218c"
      end
    
      resource "Django" do
        url "https://pypi.python.org/packages/source/D/Django/Django-1.8.tar.gz"
        sha1 "19f5468e100b5004b020b59d4f457483122b3e45"
      end
    
      resource "django-bootstrap3" do
        url "https://pypi.python.org/packages/source/d/django-bootstrap3/django-bootstrap3-5.4.0.tar.gz"
        sha1 "9c56ca0fabfa2b18a3f7a91b14bf1866343b2c80"
      end
    
      resource "django-revproxy" do
        url "https://pypi.python.org/packages/source/d/django-revproxy/django-revproxy-0.9.0.tar.gz"
        sha1 "378a51b9c1bb0b5955c8e11842f03848eecac0f8"
      end
    
        resource "django-tables2" do
        url "https://pypi.python.org/packages/source/d/django-tables2/django-tables2-0.16.0.tar.gz"
        sha1 "ef3c2e9d7fb2961cee41399607589fcb9eb06b63"
      end
    
        resource "gitdb" do
        url "https://pypi.python.org/packages/source/g/gitdb/gitdb-0.6.4.tar.gz"
        sha1 "21cbba28199802e73e0c4a83b7c02369bbf8c7d7"
      end
    
        resource "GitPython" do
        url "https://pypi.python.org/packages/source/G/GitPython/GitPython-1.0.0.tar.gz"
        sha1 "0c102b969973dd055488e519e1854094533c5cfe"
      end
    
        resource "psutil" do
        url "https://pypi.python.org/packages/source/p/psutil/psutil-2.2.1.tar.gz"
        sha1 "ddf58b3a0e699e142586b67097e3ae062766f11d"
      end
    
          resource "Pygments" do
        url "https://pypi.python.org/packages/source/P/Pygments/Pygments-2.0.2.tar.gz"
        sha1 "fe2c8178a039b6820a7a86b2132a2626df99c7f8"
      end
    
        resource "pytz" do
        url "https://pypi.python.org/packages/source/p/pytz/pytz-2015.2.tar.bz2"
        sha1 "9a1b8bf9b1222f63f7d677f124453d13d7ed27c8"
      end
    
        resource "requests" do
        url "https://pypi.python.org/packages/source/r/requests/requests-2.6.0.tar.gz"
        sha1 "ad7327c73e8be8c188ad489d511097202b1fef12"
      end
    
      resource "six" do
        url "https://pypi.python.org/packages/source/s/six/six-1.9.0.tar.gz"
        sha1 "d168e6d01f0900875c6ecebc97da72d0fda31129"
      end
    
      resource "smmap" do
        url "https://pypi.python.org/packages/source/s/smmap/smmap-0.9.0.tar.gz"
        sha1 "2aad63033aab4e4cf1a3f75f521d985da8b1353f"
      end
    
        resource "South" do
        url "https://pypi.python.org/packages/source/S/South/South-1.0.2.tar.gz"
        sha1 "8f9bf809aa8c63e5b46359d3751b76786b9f58c6"
      end
    
        resource "urllib3" do
        url "https://pypi.python.org/packages/source/u/urllib3/urllib3-1.10.1.tar.gz"
        sha1 "434ff4f51b566076ed2f41ff71f24e1b0d5c9ebd"
      end
    
      def install
        ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/python2.7/site-packages"
        %w[daemon Django django-bootstrap3 django-revproxy django-tables2 gitdb GitPython psutil Pygments pytz requests six smmap South urllib3].each do |r|
          resource(r).stage do
            system "python", *Language::Python.setup_install_args(libexec/"vendor")
          end
        end
    #
        ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python2.7/site-packages"
        system "python", *Language::Python.setup_install_args(libexec)
    #
        bin.install Dir["#{libexec}/bin/*"]
        bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
      end
    
    plist_options :manual => "python"
    
      def plist; <<-EOS.undent
        <?xml version="1.0" encoding="UTF-8"?>
        <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
        <plist version="1.0">
            <dict>
                <key>KeepAlive</key>
                <true/>
                <key>Label</key>
                <string>#{plist_name}</string>
                <key>ProgramArguments</key>
                <array>
                    <string>#{HOMEBREW_PREFIX}/bin/python</string>
                    <string>#{prefix}/manage.py</string>
                    <string>runserver</string>
                </array>
                <key>RunAtLoad</key>
                <true/>
            </dict>
        </plist>
      EOS
      end
    
      test do
        system "echo", "Done!"
      end
    end

but getting this:

    $ brew install scirius
    ==> Downloading https://github.com/StamusNetworks/scirius/archive/scirius-1.0.tar.gz
    Already downloaded: /Library/Caches/Homebrew/scirius-1.0.tar.gz
    ==> Downloading https://pypi.python.org/packages/source/d/daemon/daemon-1.1.tar.gz
    Already downloaded: /Library/Caches/Homebrew/scirius--daemon-1.1.tar.gz
    ==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/scirius/1.0/libexec/vendor --single-version-externally-managed --record=inst
    ==> Downloading https://pypi.python.org/packages/source/D/Django/Django-1.8.tar.gz
    Already downloaded: /Library/Caches/Homebrew/scirius--Django-1.8.tar.gz
    ==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/scirius/1.0/libexec/vendor --single-version-externally-managed --record=inst
    ==> Downloading https://pypi.python.org/packages/source/d/django-bootstrap3/django-bootstrap3-5.4.0.tar.gz
    Already downloaded: /Library/Caches/Homebrew/scirius--django-bootstrap3-5.4.0.tar.gz
    ==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/scirius/1.0/libexec/vendor --single-version-externally-managed --record=inst
    ==> Downloading https://pypi.python.org/packages/source/d/django-revproxy/django-revproxy-0.9.0.tar.gz
    Already downloaded: /Library/Caches/Homebrew/scirius--django-revproxy-0.9.0.tar.gz
    ==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/scirius/1.0/libexec/vendor --single-version-externally-managed --record=inst
    ==> Downloading https://pypi.python.org/packages/source/d/django-tables2/django-tables2-0.16.0.tar.gz
    Already downloaded: /Library/Caches/Homebrew/scirius--django-tables2-0.16.0.tar.gz
    ==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/scirius/1.0/libexec/vendor --single-version-externally-managed --record=inst
    ==> Downloading https://pypi.python.org/packages/source/g/gitdb/gitdb-0.6.4.tar.gz
    Already downloaded: /Library/Caches/Homebrew/scirius--gitdb-0.6.4.tar.gz
    ==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/scirius/1.0/libexec/vendor --single-version-externally-managed --record=inst
    ==> Downloading https://pypi.python.org/packages/source/G/GitPython/GitPython-1.0.0.tar.gz
    Already downloaded: /Library/Caches/Homebrew/scirius--GitPython-1.0.0.tar.gz
    ==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/scirius/1.0/libexec/vendor --single-version-externally-managed --record=inst
    ==> Downloading https://pypi.python.org/packages/source/p/psutil/psutil-2.2.1.tar.gz
    Already downloaded: /Library/Caches/Homebrew/scirius--psutil-2.2.1.tar.gz
    ==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/scirius/1.0/libexec/vendor --single-version-externally-managed --record=inst
    ==> Downloading https://pypi.python.org/packages/source/P/Pygments/Pygments-2.0.2.tar.gz
    Already downloaded: /Library/Caches/Homebrew/scirius--Pygments-2.0.2.tar.gz
    ==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/scirius/1.0/libexec/vendor --single-version-externally-managed --record=inst
    ==> Downloading https://pypi.python.org/packages/source/p/pytz/pytz-2015.2.tar.bz2
    Already downloaded: /Library/Caches/Homebrew/scirius--pytz-2015.2.tar.bz2
    ==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/scirius/1.0/libexec/vendor --single-version-externally-managed --record=inst
    ==> Downloading https://pypi.python.org/packages/source/r/requests/requests-2.6.0.tar.gz
    Already downloaded: /Library/Caches/Homebrew/scirius--requests-2.6.0.tar.gz
    ==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/scirius/1.0/libexec/vendor --single-version-externally-managed --record=inst
    ==> Downloading https://pypi.python.org/packages/source/s/six/six-1.9.0.tar.gz
    Already downloaded: /Library/Caches/Homebrew/scirius--six-1.9.0.tar.gz
    ==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/scirius/1.0/libexec/vendor --single-version-externally-managed --record=inst
    ==> Downloading https://pypi.python.org/packages/source/s/smmap/smmap-0.9.0.tar.gz
    Already downloaded: /Library/Caches/Homebrew/scirius--smmap-0.9.0.tar.gz
    ==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/scirius/1.0/libexec/vendor --single-version-externally-managed --record=inst
    ==> Downloading https://pypi.python.org/packages/source/S/South/South-1.0.2.tar.gz
    Already downloaded: /Library/Caches/Homebrew/scirius--South-1.0.2.tar.gz
    ==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/scirius/1.0/libexec/vendor --single-version-externally-managed --record=inst
    ==> Downloading https://pypi.python.org/packages/source/u/urllib3/urllib3-1.10.1.tar.gz
    Already downloaded: /Library/Caches/Homebrew/scirius--urllib3-1.10.1.tar.gz
    ==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/scirius/1.0/libexec/vendor --single-version-externally-managed --record=inst
    ==> python -c import setuptools... --no-user-cfg install --prefix=/usr/local/Cellar/scirius/1.0/libexec --single-version-externally-managed --record=installed.t
    --record=installed.txt
    
    Traceback (most recent call last):
      File "<string>", line 3, in <module>
    IOError: [Errno 2] No such file or directory: 'setup.py'
    couldn't understand kern.osversion `14.4.0'
    
    READ THIS: https://git.io/brew-troubleshooting
    
    These open issues may also help:
    creating new formula 'scirius' (rules manager for 'suricata' IDPS) (https://github.com/Homebrew/homebrew/issues/41499)
    New formula - scirius - rules manager for suricata IDPS (https://github.com/Homebrew/homebrew/pull/41538)

I've created Gist:

    $ brew gist-logs scirius
    couldn't understand kern.osversion `14.4.0'
    https://gist.github.com/18ec98947f05c1f2fb73

Please help

**Comment by Drewshg312** [comment]
Hey guys from Homebrew! Any ideas what's wrong with my formula?

**Comment by tdsmith** [comment]
You're trying to run a setup.py that doesn't exist.

**Comment by Drewshg312** [comment]
ok. there is no `setup.py` word in the formula, meaning that it's not me - it's Homebrew trying to run a **setup.py**  which for some reason doesn't exist( So how I solve this??

**Comment by Drewshg312** [comment]
How do I ask Homebrew to do this:

    git clone https://github.com/StamusNetworks/Scirius.git /usr/local/Cellar/scirius/1.0.0

**Comment by tdsmith** [comment]
`system "python", *Language::Python.setup_install_args(libexec)` invokes setup.py.

Please read the documentation in /usr/local/share/doc/homebrew.

**Comment by Drewshg312** [comment]
wow thanx! What about `git clone` command?

**Comment by Drewshg312** [comment]
There is nothing about **setup.py** in documentation except of this(which I don't even understand how to use):

    ## Brewed Python modules
    
    For brewed Python, modules installed with `pip` or `python setup.py install` will be installed to `$(brew --prefix)/lib/pythonX.Y/site-packages` directory (explained above). Executable python scripts will be in `$(brew --prefix)/bin`.
    
    The system Python may not know which compiler flags to set in order to build bindings for software installed in Homebrew so you may need to:
    
    `CFLAGS=-I$(brew --prefix)/include LDFLAGS=-L$(brew --prefix)/lib pip install <package>`.

**Comment by Drewshg312** [comment]
Does anyone have any ideas why am I getting this message:

     Traceback (most recent call last):
      File "<string>", line 3, in <module>
     IOError: [Errno 2] No such file or directory: 'setup.py'

     READ THIS: https://git.io/brew-troubleshooting

    These open issues may also help:
    creating new formula 'scirius' (rules manager for 'suricata' IDPS) (https://github.com/Homebrew/homebrew/issues/41499)
    New formula - scirius - rules manager for suricata IDPS (https://github.com/Homebrew/homebrew/pull/41538)

**Comment by Drewshg312** [comment]
Well I'm one step closer - need another hint.

I've added  `head "https://github.com/StamusNetworks/scirius.git"` So I want the git repository to be cloned to `/usr/local/Cellar/scirius/HEAD/` 

What I've done just commented the block which calls 

    system "python", *Language::Python.setup_install_args(libexec) 

because I don't need to invoke `setup.py` which doesn't exist in actual app. It should be just cloned to either `/usr/local/Cellar/scirius/HEAD` or `/usr/local/Cellar/scirius/1.0`

So when I do this:

      def install
        ENV.prepend_create_path "PYTHONPATH", libexec/"vendor/lib/python2.7/site-packages"
        %w[daemon Django django-bootstrap3 django-revproxy django-tables2 gitdb GitPython psutil Pygments pytz requests six smmap South urllib3].each do |r|
          resource(r).stage do
            system "python", *Language::Python.setup_install_args(libexec/"vendor")
          end
      end
        # ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python2.7/site-packages"
        # system "python", *Language::Python.setup_install_args(libexec)
        # bin.install Dir["#{libexec}/bin/*"]
        # bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
      end

I'm getting this in **/usr/local/Cellar/scirius/HEAD**:

    drwxr-xr-x  7 Andryuwka  admin   238B Jul 14 23:33 ./
    -rw-r--r--  1 Andryuwka  admin   275B Jul 14 23:33 INSTALL_RECEIPT.json
    -rw-r--r--  1 Andryuwka  admin   584B Jul 14 23:33 homebrew.mxcl.scirius.plist
    drwxr-xr-x  5 Andryuwka  admin   170B Jul 14 23:33 libexec/
    drwxr-xr-x  3 Andryuwka  admin   102B Jul 14 23:32 ../
    -rw-r--r--  1 Andryuwka  wheel    34K Apr 18 01:15 LICENSE
    -rw-r--r--  1 Andryuwka  wheel    10K Apr 18 01:15 README.rst

But should be this (the contain of repository):

    drwxr-xr-x  16 Andryuwka  admin   544B Jul 14 22:14 ./
    -rw-r--r--   1 Andryuwka  admin    13M Jul 14 22:14 db.sqlite3
    -rw-r--r--   1 Andryuwka  admin   249B Jul  9 19:26 requirements.txt
    -rw-r--r--   1 Andryuwka  admin   145B Jul  9 19:26 requirements_new.txt
    drwxr-xr-x  18 Andryuwka  admin   612B Jul  9 00:21 scirius/
    drwxr-xr-x  12 Andryuwka  admin   408B Jul  8 19:15 .git/
    drwxr-xr-x   4 Andryuwka  admin   136B Jul  6 16:35 git-sources/
    drwxr-xr-x  16 Andryuwka  admin   544B Jul  6 16:07 accounts/
    drwxr-xr-x  27 Andryuwka  admin   918B Jul  6 16:07 rules/
    drwxr-xr-x  19 Andryuwka  admin   646B Jul  6 16:07 suricata/
    -rw-r--r--   1 Andryuwka  admin    52B Jul  6 16:04 .gitignore
    -rw-r--r--   1 Andryuwka  admin    34K Jul  6 16:04 LICENSE
    -rw-r--r--   1 Andryuwka  admin    10K Jul  6 16:04 README.rst
    drwxr-xr-x   3 Andryuwka  admin   102B Jul  6 16:04 doc/
    -rw-r--r--   1 Andryuwka  admin   250B Jul  6 16:04 manage.py
    drwxr-xr-x   8 Andryuwka  admin   272B Jul  6 16:04 ../

So what should I do to get the `head` repository cloned to **/usr/local/Cellar/scirius/HEAD** or **/usr/local/Cellar/scirius/1.0**?

**Comment by DomT4** [comment]
Let's keep the discussion in the PR #41538.

**Comment by Drewshg312** [comment]
Ok! Sorry I've recently created a new issue... Please help me with this... It just freaks me out( lol
